### PR TITLE
btop: Add darwin support

### PR DESCRIPTION
**Pull request by Steven0351**
###### Motivation for this change
Adds darwin support for btop

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/system/btop/default.nix [hunk 1]
```diff
@@ -14,14 +17,36 @@ stdenv.mkDerivation rec {
...
+
+  ADDFLAGS = with darwin.apple_sdk;
+    lib.optional stdenv.isDarwin
+      "-F${frameworks.IOKit}/Library/Frameworks/";
+
+  buildInputs = with darwin.apple_sdk;
+    lib.optionals stdenv.isDarwin [
+      frameworks.CoreFoundation
+      frameworks.IOKit
+    ] ++ lib.optional (stdenv.isDarwin && !stdenv.isAarch64) (
```
Comment (prusnak) [inline pkgs/tools/system/btop/default.nix:32]: This is more in-line with what you want to say:
```suggestion
    ] ++ lib.optional (stdenv.isDarwin && stdenv.isx86_64) (
```

**Review by prusnak — CHANGES_REQUESTED** [review]

File: pkgs/tools/system/btop/default.nix [hunk 2]
```diff
@@ -14,14 +17,36 @@ stdenv.mkDerivation rec {
...
+  hardeningDisable = lib.optionals (stdenv.isAarch64 && stdenv.isDarwin) [ "stackprotector" ];
+
+  ADDFLAGS = with darwin.apple_sdk;
+    lib.optional stdenv.isDarwin
+      "-F${frameworks.IOKit}/Library/Frameworks/";
+
+  buildInputs = with darwin.apple_sdk;
+    lib.optionals stdenv.isDarwin [
+      frameworks.CoreFoundation
+      frameworks.IOKit
```
Comment (prusnak) [inline pkgs/tools/system/btop/default.nix:31]:
```suggestion
  buildInputs = with darwin.apple_sdk.frameworks;
    lib.optionals stdenv.isDarwin [
      CoreFoundation
      IOKit
```

**Review by prusnak — CHANGES_REQUESTED** [review]

File: pkgs/tools/system/btop/default.nix (hunk 2 above)
Comment (Steven0351) [reply pkgs/tools/system/btop/default.nix:31]: @prusnak Changing this here is going to make line 38 to resolve

File: pkgs/tools/system/btop/default.nix [hunk 3]
```diff
@@ -14,14 +17,36 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-uKR1ogQwEoyxyWBiLnW8BsOsYgTpeIpKrKspq0JwYjY=";
   };
 
+  nativeBuildInputs = lib.optionals stdenv.isDarwin [ gcc11 ];
```
Comment (risicle) [inline pkgs/tools/system/btop/default.nix:20]: When a darwin package requires gcc we tend to achieve this by switching out the `stdenv` for a `gccStdenv` e.g. `gcc11Stdenv`. Supplying gcc as an additional input results in a funny environment that has access to both compilers. See usages of `gccStdenv` in `all-packages.nix` for examples.

Comment (Steven0351) [reply pkgs/tools/system/btop/default.nix:20]: So, in my ignorance in how nixpkgs works, I'm actually seeing in all-packages.nix that stdenv for btop is already being set to gcc11Stdenv. I based the darwin implementation off of what I came up with for a local package I have set up to use btop on my machines, which of course was not setting stdenv to gcc11Stdenv. Thank you for pointing this out, I never would have though to look there. Will be updating the PR to remove that altogether.

**Review by risicle — APPROVED** [review]
WFM macos 10.15 & nixos x86_64

**Pull request merged**